jobs/ocp4/pds-ocp4-install.yaml

```diff
@@ -0,0 +1,258 @@
+        - string:
+            name: REGISTRY_NAMESPACE
+            default: 'integreatly'
+            description: "Registry name in quay.io"
+        - choice:
+            name: INSTALLATION_TYPE
+            description: "Installation type"
+            choices:
+              - workshop
+              - managed
+        - string:
+            name: NAMESPACE_PREFIX
+            description: "Prefix of Integreatly Operator namespace, defauls to `BUILD_NUMBER`"
+        - bool:
+            name: CLEANUP_AFTERWARD
+            default: true
+            description: "Indicates whether to cleanup after execution"
+        - bool:
+            name: CLEANUP_BEFOREHAND
+            default: true
+            description: "Indicates whether to cleanup before execution"
+        - string:
+            name: NAMESPACE_CLEANUP_BEFOREHAND
+            description: "Namespace to be removed before the Integreatly Operator deploy. Can be left empty"
+        - string:
+            name: TEST_SUITES_REPOSITORY
+            default: 'https://gitlab.cee.redhat.com/integreatly-qe/integreatly-qe.git'
+            description: "Repository of Integreatly test suites, YAML templates are stored there"
+        - string:
+            name: TEST_SUITES_BRANCH
+            default: 'development'
+            description: "Branch of Integreatly test suites repository"
+        - string:
+            name: RECIPIENTS
+            default: 'integreatly-qe@redhat.com'
+            description: "Whitespace- or comma-separated list of recipient addresses"
+    dsl: |
+        Boolean cleanupAfterward = params.CLEANUP_AFTERWARD
+        Boolean cleanupBeforehand = params.CLEANUP_BEFOREHAND
+        Boolean selfSignedCerts = params.SELF_SIGNED_CERTS
+
+        String registryNamespace = 'integreatly'
+        if (params.REGISTRY_NAMESPACE) {
+            registryNamespace = params.REGISTRY_NAMESPACE
+        }
+
+        String clusterAPI = "https://api.cluster-${RHPDS_CITY}.${RHPDS_CITY}.openshiftworkshop.com:6443"
+
+        // by default build number is used as namespacePrefix to prevent clashes with previous Integreatly Operator deployments
+        String namespacePrefix = BUILD_NUMBER
+        if (NAMESPACE_PREFIX) {
+            namespacePrefix = NAMESPACE_PREFIX
+        }
+        String namespaceName = namespacePrefix + '-integreatly-operator'
+
+        def err = null
+
+        node('cirhos_rhel7') {
+            timeout(30) { ansiColor('gnome-terminal') { timestamps {
+                try {
+
+                    stage('Install oc cli version 4') {
+                        withCredentials([string(credentialsId: "${ADMIN_PASSWORD_CREDENTIALS}", variable: 'ADMIN_PASSWORD')]) {
+                            sh(
+                                returnStdout: false,
+                                script: """
+                                    wget -q https://mirror.openshift.com/pub/openshift-v4/clients/ocp/4.1.8/openshift-client-linux-4.1.8.tar.gz
```

Comment:
I think we should update our Jenkins template in the future to also have available the oc 4 client. We can create alias for that - e.g `oc4`

Reply:
yep, there are definitely better ways of doing this. So yes, having oc4 installed on Jenkins slaves is definitely better. I would think of something along the lines of creating a symlink at the very beginning so that `oc` would be redirected to oc version 4 (and in the code you can just use `oc` instead of `oc4`. I will create a jira for this.